src/reuse/header.py

```diff
@@ -351,19 +351,26 @@ def _verify_paths_line_handling(
 
 
 def _verify_paths_comment_style(paths: List[Path], parser: ArgumentParser):
+    # Create empty list for files whose comment style is unknown
+    unrecognised_files = []
+
     for path in paths:
         style = _get_comment_style(path)
         not_uncommentable = not _is_uncommentable(path)
 
         # TODO: This check is duplicated.
         if style is None and not_uncommentable:
-            parser.error(
-                _(
-                    "'{path}' does not have a recognised file extension,"
-                    " please use --style, --explicit-license or"
-                    " --skip-unrecognised"
-                ).format(path=path)
-            )
+            unrecognised_files.append(path)
+
+    # If at least one file was unrecognised, print error
+    if len(unrecognised_files) > 0:
```

Comment:
Lists in Python are truthy :) The conventional way to do this is `if unrecognised_files`